lib/mongo_ecto/normalized_query.ex

```diff
@@ -78,13 +78,10 @@ defmodule Mongo.Ecto.NormalizedQuery do
         {:skip,  value} -> ["$skip":  value]
       end)
       |> Kernel.++(pipeline)
+      |> (fn pipeline -> if query != %{}, do: [["$match": query] | pipeline], else: pipeline end).()
```

Comment:
Let's extract it into a private function, anonymous functions in the pipeline like this are not a great idea in terms of readability.